plugins/modules/dcnm_image_upgrade.py

```diff
@@ -0,0 +1,1400 @@
+            ):
+                upgrade_devices.append(switch)
+
+        self._stage_images(stage_devices)
+        self._validate_images(validate_devices)
+
+        self._verify_install_options(upgrade_devices)
+        self._upgrade_images(upgrade_devices)
+
+    def handle_deleted_state(self) -> None:
+        """
+        Delete the image policy from the switch(es)
+
+        Caller: main()
+        """
+        msg = "ENTERED"
+        self.log.debug(msg)
+
+        self._attach_or_detach_image_policy("detach")
+
+    def handle_query_state(self) -> None:
+        """
+        Return the ISSU state of the switch(es) listed in the playbook
+
+        Caller: main()
+        """
+        instance = SwitchIssuDetailsByIpAddress(self.ansible_module)
+        instance.refresh()
+        response_current = copy.deepcopy(instance.response_current)
+        if "DATA" in response_current:
+            response_current.pop("DATA")
+        self.task_result.response_issu_status = copy.deepcopy(response_current)
+        self.task_result.response = copy.deepcopy(response_current)
+        for switch in self.need:
+            instance.filter = switch.get("ip_address")
+            msg = f"SwitchIssuDetailsByIpAddress.filter: {instance.filter}, "
+            msg += f"SwitchIssuDetailsByIpAddress.filtered_data: {json.dumps(instance.filtered_data, indent=4, sort_keys=True)}"
+            self.log.debug(msg)
+            if instance.filtered_data is None:
+                continue
+            self.task_result.diff_issu_status = instance.filtered_data
+            self.task_result.diff = instance.filtered_data
+
+    def _failure(self, resp) -> None:
+        """
+        Caller: self.attach_policies()
+        """
+        res = copy.deepcopy(resp)
+
+        if resp.get("DATA"):
+            data = copy.deepcopy(resp.get("DATA"))
+            if data.get("stackTrace"):
+                data.update(
+                    {"stackTrace": "Stack trace is hidden, use '-vvvvv' to print it"}
+                )
+                res.update({"DATA": data})
+
+        self.ansible_module.fail_json(msg=res)
+
+
+def main():
+    """main entry point for module execution"""
+
+    element_spec = {
+        "config": {"required": True, "type": "dict"},
+        "state": {"default": "merged", "choices": ["merged", "deleted", "query"]},
+    }
+
+    ansible_module = AnsibleModule(argument_spec=element_spec, supports_check_mode=True)
+
+    # Create the base/parent logger for the dcnm collection.
+    # To enable logging, set enable_logging to True.
+    # log.config can be either a dictionary, or a path to a JSON file
+    # Both dictionary and JSON file formats must be conformant with
+    # logging.config.dictConfig and must not log to the console.
+    # For an example configuration, see:
+    # $ANSIBLE_COLLECTIONS_PATH/cisco/dcnm/plugins/module_utils/common/logging_config.json
+    enable_logging = False
```

Comment:
This should be changed to use environment variable as was done in dcnm_fabric.py.  For now, since enable_logging is False, it'll be fine.  I've made a reminder for myself to fix this later.